### ISSUE TITLE
altis/dev-tools as a dependency causes unnecessary non-dev dependencies

### ISSUE THREAD

**Issue by johnbillion** [humanmade/altis#9]
If you follow [the basic Getting Started instructions for an Altis project](https://www.altis-dxp.com/resources/docs/getting-started/) you end up with a bunch of developer libraries in your Composer dependency graph, even when installing the site for production use with `composer install --non-dev`.

This includes `phpunit/phpunit`, `wp-phpunit/wp-phpunit`, and others.

The reason is that this `altis/altis` meta package has `altis/dev-tools` as a non-dev dependency, which in turn has a non-dev dependency on `phpunit/phpunit` and `wp-phpunit/wp-phpunit`.

I think the problem is that `altis/dev-tools` should be a dev dependency rather than a dependency.

Thoughts?

**Comment by johnbillion** [comment]
Having given it further thought, I think the problem is that the altis/dev-tools package is a combination of stuff that you want deployed (debugging plugins) and stuff that you don't (PHPUnit) but they're packaged together. Maybe altis/dev-tools needs to be split into two packages?

**Comment by roborourke** [comment]
@johnbillion yeah that's a difficult one because dependencies can't install their dev dependencies when you run the install from your root project. It's not ideal to have phpunit installed in the production environment I agree.

There are a few options we could consider here:

- Add handling to the core installer to install a module's dev dependencies when not carrying out a no-dev install
- Similar to the above but have modules define which dependencies should be ignored in a production install
- Allow for a second step in module installation that installs additional dependencies if no-dev isn't specified using `composer install <package>, ...` from the root so the `composer.json` isn't updated

I think the first one might be the best option, or along those lines.

Can you elaborate more on how an additional package would solve the issue? Another meta package of dev dependencies like this?

```json
{
   "require": {
      "altis/altis": "^3.0.0"
   },
   "require-dev": {
     "altis/dev": "^3.0.0"
   }
}
```

I'm more in favour of whatever solution keeps things easier to install.

**Comment by rmccue** [comment]
Apart from PHPUnit, are there any others?

**Comment by roborourke** [comment]
Nope not right now
